envoy-control-services/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/services/transformers/InvalidPortFilterTest.kt

```diff
@@ -0,0 +1,136 @@
+package pl.allegro.tech.servicemesh.envoycontrol.services.transformers
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import pl.allegro.tech.servicemesh.envoycontrol.services.ServiceInstance
+import pl.allegro.tech.servicemesh.envoycontrol.services.ServiceInstances
+
+internal class InvalidPortFilterTest {
+
+    @Test
+    fun `should filter out instances with port setup as zero`() {
+        // given
+        val filter = InvalidPortFilter()
+        val instances = sequenceOf(
+            ServiceInstances(
```

Comment:
To do not mess up the test, I would use some builder/mother/creator/helper class to create this objects. Or at least maybe create separate methods for it.

Reply:
I'd rather not, I think test should be easy to read, and I see no value of extracting constructor objects to another method which has to take params to just return constructed object as it would look very similliar

Reply:
but we could make some default parameters and ignore most of constructor properties wich are just a noise in this test case (ex. regular or canary property).

Reply:
simplified

Reply:
The value of extracting is having a simple given-when-then - as a reader I can easily see what the test is testing. So, for me, even extracting whole creation (without parameterizationing) to method is better than creating it in test.

Reply:
I would not agree with that cuz I'd like to understand test on my first sight of it